### PR TITLE
feat: add synth_node_bin

### PR DESCRIPTION
**Pull request by Rqnsom**
Two things are here:
```
    feat: add synth_node_bin

    Synth node binary can be used to start a synth_node from an external
    binary and connect it to any node via the network protocol.
```
```
    chore: rename package name to a more specific name

    Because the term "ziggurat" is a more general term these days,
    there are multiple ziggurat projects so ZCash implementation of
    the Ziggurat test suite is not unique anymore.
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "ziggurat"
+name = "ziggurat-zcash"
```
Comment (zeapoz) [inline Cargo.toml:2]: Good call changing this!

**Review by zeapoz — APPROVED** [review]
LGTM :+1:

File: synth_node_bin/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        fmt()
+            .with_test_writer()
+            .with_env_filter(EnvFilter::from_default_env())
+            .init();
+    }
+
+    loop {
+        println!("Starting a synthetic node.");
+
+        if let Err(e) = start_synth_node(node_addr).await {
```
Comment (kylegranger) [inline synth_node_bin/src/main.rs:62]: I'd call this `run_synth_node` or `do_synth_node` -- you're doing many more things here than starting a synth node.

**Review by kylegranger — APPROVED** [review]
🚀  LGTM!

File: synth_node_bin/src/main.rs (hunk 2 above)
Comment (Rqnsom) [reply synth_node_bin/src/main.rs:62]: Good point! Thank you!

Comment (Rqnsom) [reply synth_node_bin/src/main.rs]: Done

**Pull request merged**